src/common/types/hyperloglog.cpp

```diff
@@ -69,4 +72,205 @@ unique_ptr<HyperLogLog> HyperLogLog::Merge(HyperLogLog logs[], idx_t count) {
 	return unique_ptr<HyperLogLog>(new HyperLogLog((void *)new_hll));
 }
 
-} // namespace duckdb
\ No newline at end of file
+idx_t HyperLogLog::GetSize() {
+	return duckdb_hll::get_size();
+}
+
+data_ptr_t HyperLogLog::GetPtr() const {
+	return (data_ptr_t)((duckdb_hll::robj *)hll)->ptr;
+}
+
+unique_ptr<HyperLogLog> HyperLogLog::Copy() const {
+	auto result = make_unique<HyperLogLog>();
+	memcpy(result->GetPtr(), GetPtr(), GetSize());
+	D_ASSERT(result->Count() == Count());
+	return result;
+}
+
+void HyperLogLog::Serialize(FieldWriter &writer) const {
+	writer.WriteBlob(GetPtr(), GetSize());
```

Comment:
Alternatively we could leave it as `dense` for now, but add an enum prior to the serialization (`HLLStorageType` or so) that would allow us to add alternative HLL storage formats without breaking backwards compatibility.

Reply:
The Redis HLL that we are using can convert sparse to dense, but not dense to sparse. We can sacrifice accuracy for size some more if we want.

With `P` the number of registers, we get the following:
```
accuracy (in %): 1.04 / sqrt(1 << P) * 100
size (in bytes): 20 + ((1 << P) * 6 + 7) / 8
```

For `P = 12` we get an accuracy of 1.625 and a size of ~3.1K, whereas for `P = 10`, we get an accuracy of 3.25 and a size of 0.8K.

Maybe the HLL can also be compressed?

Reply:
Yes, that is exactly what I was getting at. If you store a` HLLStorageType` enum as the first field, you can read the subsequent blob and decompress it based on the storage type. For now we could only support a single storage type (`HLLStorageType::UNCOMPRESSED`) but we could add compression later on.